### DISCUSSION TITLE
Unset a property that is set by a style

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [AvaloniaUI/Avalonia#16627] [Q&A] [answered]
How can I unset a property that is set by a style?

Background: Fluent theme sets the height of `Menu` to a fixed height of 32. This does not take the font size into account so that the text gets cropped. I want that the height of the `Menu` automatically adopts to the current font size.

**Comment by stevemonaco** [comment, marked as answer]
You can't unset values in styles, AFAIK. I'm not sure that makes sense because the property system heavily relies on fallbacks. A single property can have many setters with varying priorities.

What you can do is instead look up and set the default value for the property. For `Height`, this is `double.NaN`. eg.

```xml
<Window ...
    xmlns:sys="using:System">

    <Window.Styles>
        <Style Selector="MenuItem">
            <Setter Property="FontSize" Value="48" />
        </Style>
        <Style Selector="Menu">
            <Setter Property="Height" Value="{x:Static sys:Double.NaN}" />
        </Style>
    </Window.Styles>

    <Grid RowDefinitions="auto,*">
        <Menu>
            <MenuItem Header="File">
                <MenuItem Header="Open..." />
            </MenuItem>
            <MenuItem Header="Edit" />
            <MenuItem Header="Help" />
        </Menu>
        <TextBlock Grid.Row="1" Text="Testing" />
    </Grid>
</Window>
```

**Reply by DevAM-Tools** [reply]
Awesome! Thank you very much! Setting the `Height` of `Menu` to `Double.NaN` did the job.
